src/ORM/RulesChecker.php

```diff
@@ -113,6 +114,26 @@ public function existsIn($field, $table, $message = null)
     }
 
     /**
+     * Returns a callable that can be used as a rule for checking that the entity is not
+     * referenced by any row in the specified table
+     *
+     * This is useful for enforcing foreign key integrity checks. (i.e.: ON DELETE RESTRICT)
+     *
+     * ### Example:
+     *
+     * ```
+     * $rules->add($rules->isNotReferencedBy('Articles'));
+     * ```
+     *
+     * @param string $table The table name to check for references
+     * @return callable
+     */
+    public function isNotReferencedBy($table)
+    {
+        return $this->_addError(new IsNotReferencedBy($table), '_isNotReferencedBy', compact('errorField', 'message'));
```

Comment:
@markstory yeah, I just copied the code from the other rules, and wasn't quite sure what this part did, and how it did it